src/pycram/designators/action_designator.py

```diff
@@ -933,3 +924,187 @@ def ground(self) -> Action:
+                adjusted_slice_pose.orientation.w = oriM[3]
+
+                # Adjust the position of the object pose by grasp in MAP
+                lift_pose = adjusted_slice_pose.copy()
+                lift_pose.pose.position.z += 2 * obj_height
+                # Perform the motion for lifting the tool
+                BulletWorld.current_bullet_world.add_vis_axis(lift_pose)
+                MoveTCPMotion(lift_pose, self.arm).resolve().perform()
+                # Perform the motion for cutting the object
+                BulletWorld.current_bullet_world.add_vis_axis(adjusted_slice_pose)
+                MoveTCPMotion(adjusted_slice_pose, self.arm).resolve().perform()
+                # Perform the motion for lifting the tool
+                BulletWorld.current_bullet_world.add_vis_axis(lift_pose)
+                MoveTCPMotion(lift_pose, self.arm).resolve().perform()
+
+        def to_sql(self) -> ORMCuttingAction:
+            """
+            Convert the action to a corresponding SQL representation for storage.
+            """
+            return ORMCuttingAction(self.arm, self.technique, self.slice_thickness)
+
+        def insert(self, session: sqlalchemy.orm.session.Session, **kwargs):
+            """
+            Insert the cutting action into the database session.
+            """
+            # insert related objects
+            object_to_be_cut = self.object_to_be_cut_at_execution.insert(session)
+            tool = self.tool_at_execution.insert(session)
+
+            action = super().insert(session)
+            action.object_to_be_cut_id = object_to_be_cut.id
+            action.tool_id = tool.id
+
+            # Additional logic for inserting cutting action data goes here
+            session.add(action)
+            session.commit()
+
+            return action
+
+    def __init__(self, object_to_be_cut: ObjectDesignatorDescription,
+                 tool: ObjectDesignatorDescription, arms: List[str], technique: Optional[str] = None):
+        """
+        Initializes a CuttingAction with specified object and tool designators, arms, and an optional cutting technique.
+
+        Args:
+            object_to_be_cut (ObjectDesignatorDescription): Designator for the object to be cut.
+            tool (ObjectDesignatorDescription): Designator for the cutting tool.
+            arms (List[str]): List of possible arms to be used for the cutting action.
+            technique (Optional[str]): Optional cutting technique to be used.
+        """
+        super(CuttingAction, self).__init__()
+        self.object_to_be_cut: ObjectDesignatorDescription = object_to_be_cut
+        self.tool: ObjectDesignatorDescription = tool
+        self.arms: List[str] = arms
+        self.technique: Optional[str] = technique
+
+    def __iter__(self):
+        """
+        Iterator for generating all possible action combinations based on the provided object, tool, and arms.
+
+        Yields:
```

Comment:
check the documentation builds if this renders correctly because of inconsistent indentation

Reply:
fixed